tests/test_dimension.py

```diff
@@ -62,6 +82,44 @@ def test_subdim_middle(self):
         assert np.all(u.data[1, -1, :, :] == 1)
         assert np.all(u.data[1, 1:3, :, :] == 2)
 
+    def test_bcs(self):
+        """
+        Tests application of an Operator consisting of multiple equations
+        defined over different sub-regions, explicitly created through the
+        use of :class:`SubDimension`s.
+        """
+        grid = Grid(shape=(20, 20))
+        x, y = grid.dimensions
+        t = grid.stepping_dim
+        thickness = 4
+
+        u = TimeFunction(name='u', save=None, grid=grid, space_order=0, time_order=1)
+
+        xleft = SubDimension.left(name='xleft', parent=x, thickness=thickness)
+        xi = SubDimension.middle(name='xi', parent=x,
+                                 thickness_left=thickness, thickness_right=thickness)
+        xright = SubDimension.right(name='xright', parent=x, thickness=thickness)
+
+        yi = SubDimension.middle(name='yi', parent=y,
+                                 thickness_left=4, thickness_right=4)
```

Comment:
the assertions below use `thickness` rather than 4 for this dimension as well - doesn't really matter but should be consistent.